OpenRA.Mods.Common/Traits/FireWarheadOnDeath.cs

```diff
@@ -21,7 +21,7 @@ public enum ExplosionType { Footprint, CenterPosition }
 	public enum DamageSource { Self, Killer }
 
 	[Desc("This actor explodes when killed.")]
-	public class ExplodesInfo : ConditionalTraitInfo, Requires<IHealthInfo>
+	public class FireWarheadOnDeathInfo : ConditionalTraitInfo, Requires<IHealthInfo>
```

Comment:
I don't want to be overly picky, however looking at the yaml changes this does sound a bit weird. We have Fire _Warhead_ OnDeath and then define `Weapons` below it. Maybe this should be named FireWeaponOnDeath?

Reply:
Well, it's not firing the projectile, so it needs to differentiate itself from normal weapon behaviour. The naming also matches `FireWarheads`

Reply:
Hm ok. Can we update the description of this trait while we are here?

Reply:
hmm, I'm not sure to what. I think the description is good enough

Reply:
"This actor fires a warhead when killed."?

Reply:
though, the current description doesn't encapsulate all functionality as well

Reply:
The current description is referencing the old trait name.

Reply:
I don't see how